ddtrace/propagation/http.py

```diff
@@ -483,6 +488,103 @@ def _extract(headers):
+
+        value = 00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01
+        base16(version) = 00
+        base16(trace-id) = 4bf92f3577b34da6a3ce929d0e0e4736
+        base16(parent-id) = 00f067aa0ba902b7
+        base16(trace-flags) = 01  // sampled
+
+    Implementation details:
+
+      - Datadog Trace and Span IDs are 64-bit unsigned integers.
+      - The W3C Trace Context Trace ID is a 16-byte hexadecimal string. This is
+        transformed for propagation between Datadog by taking the lower
+        8-bytes.
+    """
+
+    @staticmethod
+    def _inject(span_context, headers):
+        # type: (Context, Dict[str, str]) -> None
+        # use hex to convert back, the trace id will be pre-pended with a bunch of 0s to fill in for previous values
+        trace_id = span_context.trace_id
+        span_id = span_context.span_id
+        if trace_id is None or span_id is None:
+            log.debug("tried to inject invalid context %r", span_context)
+            return
+
+        sampling_priority = span_context.sampling_priority
+        trace_flags = 1 if sampling_priority and sampling_priority >= AUTO_KEEP else 0
+
+        # There is currently only a single version so we always start with 00
+        traceparent = "00-{:032x}-{:016x}-{:02x}".format(trace_id, span_id, trace_flags)
+        headers[_HTTP_HEADER_W3C_TRACEPARENT] = traceparent
+
+    @staticmethod
+    def _extract(headers):
+        # type: (Dict[str, str]) -> Optional[Context]
+        tp = _extract_header_value(_POSSIBLE_HTTP_HEADER_W3C_TRACEPARENT, headers)
+        if tp is None:
+            return None
+
+        version, trace_id_hex, span_id_hex, trace_flags_hex = tp.split("-")
+
+        try:
+            # currently 00 is the only version format, but if future versions come up we may need to add changes
+            if version != "00":
+                log.warning("unsupported traceparent version:%s . Will still attempt to parse.", (version))
+
+            if len(trace_id_hex) == 32 and len(span_id_hex) == 16 and len(trace_flags_hex) >= 2:
+                trace_id = _hexadec_id_to_dd_id(trace_id_hex)
+                span_id = _hexadec_id_to_dd_id(span_id_hex)
+
+                # All 0s are invalid values
+                assert trace_id != 0
+                assert span_id != 0
+
+                trace_flags = _hexadec_id_to_dd_id(trace_flags_hex)
+                # there's currently only one trace flag, which denotes sampling priority was set to keep
+                sampling_priority = float(trace_flags & 0x1)
+
+            else:
+                log.debug("W3C traceparent hex length incorrect%s", (tp))
+
+            return Context(
+                trace_id=trace_id,
+                span_id=span_id,
+                sampling_priority=sampling_priority,
+            )
+        except (ValueError, AssertionError) as e:
+            log.warning("received invalid w3c traceparent: %s. Error: %s", (tp, e))
```

Comment:
🤔  pardon my `logger` ignorance, but shouldn't this be
```suggestion
            log.warning("received invalid w3c traceparent: %s. Error: %s", tp, e)
```
?